packages/os/network-config.service

```
@@ -0,0 +1,20 @@
+[Unit]
+Description=Bottlerocket network configuration system
+Requires=var-lib-bottlerocket.mount
```

Nit: Maybe a note here saying that the mount should be there because the service will attempt to read from it? Also in other units we use this when we need a mount to be present:

```suggestion
RequiresMountsFor=/var/lib/bottlerocket
```